libs/server/Module/ModuleRegistrar.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace Garnet.server.Module
+{
+    /// <summary>
+    /// Common interface that all Garnet modules must implement.
+    /// </summary>
+    public interface IModule
+    {
+        /// <summary>
+        /// Called when the module is loaded.
+        /// </summary>
+        /// <param name="context">Module load context</param>
+        /// <param name="moduleArgs">Module load args</param>
+        void OnLoad(ModuleLoadContext context, List<string> moduleArgs);
```

Comment:
Should be `string[] args` - zero-length if there are no args in the common case.